content/blog/2025/laravel-performance-monitoring-complete-apm-comparison-guide.md

```
@@ -0,0 +1,2399 @@
+        // Database query monitoring
+        DB::listen(function ($query) use ($statsd) {
+            $statsd->timing('database.query.time', $query->time);
+            $statsd->increment('database.query.count', 1, [
+                'connection' => $query->connectionName
+            ]);
+        });
+
+        // Cache monitoring
+        Event::listen('cache.*', function ($event) use ($statsd) {
+            $statsd->increment('cache.' . $event->type, 1, [
+                'key' => $event->key
```

markdownlint-cli2 (0.18.1)

Emphasis used instead of a heading

(MD036, no-emphasis-as-heading)